calc/l3_route_resolver.go

```diff
@@ -343,7 +343,17 @@ func (c *L3RouteResolver) OnResourceUpdate(update api.Update) (_ bool) {
 				Addr: ip.FromCalicoIP(*ipv4).(ip.V4Addr),
 				CIDR: ip.CIDRFromCalicoNet(*caliNodeCIDR).(ip.V4CIDR),
 			}
+		} else if len(node.Spec.Addresses) > 0 {
+			ipv4, caliNodeCIDR, err := cnet.ParseCIDROrIP(node.Spec.Addresses[0].Address)
```

Comment:
Will the `0` address always be the right one?  I thought the logic in libcalico-go was more complex to find the "right" IP?

Worth commenting here that we're mainly doing this to be consistent with the host metadata IP so that BPF mode and iptables mode behave the same in this corner case.